include/linear_feedback_controller/robot_model_builder.hpp

```diff
@@ -0,0 +1,128 @@
+#ifndef LINEAR_FEEDBACK_CONTROLLER_ROBOT_MODEL_BUILDER_HPP
+#define LINEAR_FEEDBACK_CONTROLLER_ROBOT_MODEL_BUILDER_HPP
+
+#include <string>
+#include <vector>
+
+// Rigid body dynamics
+#include <pinocchio/fwd.hpp>
+#include <pinocchio/multibody/model.hpp>
+
+#include "pinocchio/multibody/data.hpp"
+
+namespace linear_feedback_controller {
+
+class RobotModelBuilder {
+ public:
+  RobotModelBuilder();
+
+  virtual ~RobotModelBuilder();
+
+  /**
+   * @brief
+   *
+   * @param in_urdf Content of the urdf file.
+   * @param in_srdf Content of the srdf file.
+   * @param in_moving_joint_names List of active joint name.
+   * @param in_robot_has_free_flyer If the robot has a free flyer or not.
+   */
+  bool build_model(const std::string& in_urdf, const std::string& in_srdf,
+                   const std::vector<std::string>& in_moving_joint_names,
+                   bool in_robot_has_free_flyer,
+                   const std::vector<std::string>& in_controlled_joint_names);
+
+  /**
+   * @brief Parse the joint moving names given by the user and build the
+   * rigid body models accordingly.
+   *
+   * @param in_moving_joint_names
+   * @param moving_joint_names
+   * @param moving_joint_ids
+   * @param locked_joint_ids
+   */
+  bool parseMovingJointNames(
+      const std::vector<std::string>& in_moving_joint_names,
+      const std::vector<std::string>& in_controlled_joint_names,
+      std::vector<std::string>& moving_joint_names,
+      std::vector<pinocchio::Index>& moving_joint_ids,
+      std::vector<pinocchio::Index>& locked_joint_ids);
```

Comment:
This should be private and accept `in_robot_has_free_flyer` so there is no need for the in-class variable